lib/util/http/index.ts

```diff
@@ -214,12 +214,23 @@ export class Http<Opts extends HttpOptions = HttpOptions> {
     // istanbul ignore else: no cache tests
     if (!resPromise) {
       if (httpOptions.repoCache) {
-        const cachedEtag = getCache().httpCache?.[url]?.etag;
-        if (cachedEtag) {
-          logger.debug(`Using cached etag for ${url}`);
+        const responseCache = getCache().httpCache?.[url];
+        // Prefer If-Modified-Since over If-None-Match
+        if (responseCache?.['lastModified']) {
+          logger.debug(
+            `http cache: trying cached Last-Modified "${responseCache?.['lastModified']}" for ${url}`,
+          );
+          options.headers = {
+            ...options.headers,
+            'If-Modified-Since': responseCache['lastModified'],
```

Comment:
Don't we wanna save when we have taken the last look at that URL?

Reply:
I think we do. This part is when we're crafting the cached request later

Reply:
See line 284